### PR TITLE
Player 2 confirm before populate board and rspec tests

### PR DESCRIPTION
**Pull request by jdoleiv**
…spec tests still not working

For some reason the Rspec tests will not populate the board.

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def update
     if @game.valid?
       current_user.games << game
       redirect_to @game
+      @game.populate_board if @game.pieces.empty?
```
Comment (jimxlin) [inline app/controllers/games_controller.rb]: Not sure if this is the cause, but you should move this line to right above `redirect_to @game`. You want to populate the board before sending the user to the page that displays the board.

File: spec/models/game_spec.rb [hunk 2]
```diff
@@ -63,4 +63,12 @@
       end
     end
   end
+
+ describe "#populate_board" do
+    it "should populate board only if both black and white player ids are present" do
+      game = Game.create(white_player_id: 0, black_player_id: 1)
```
Comment (jimxlin) [inline spec/models/game_spec.rb]: Where are you getting the values `0` and `1`? These have to be ids that belong to users that are actually in the database.

File: app/controllers/games_controller.rb (hunk 1 above)
Comment (jdoleiv) [reply app/controllers/games_controller.rb]: Hmm, its still not happy...

File: spec/models/game_spec.rb (hunk 2 above)
Comment (jacobwgillespie) [reply spec/models/game_spec.rb]: My guess is that this is why the tests are failing - the piece classes have a validation that ensures that their user exists.

You should probably add something like:

```ruby
white_user = User.create(... dummy details ...)
black_user = User.create(... dummy details ...)
game = Game.create(white_player_id: white_player.id, black_player_id: black_player.id)
```

Comment (jdoleiv) [reply spec/models/game_spec.rb]: Thanks Jacob.  Jim helped me get to the user creation part, but I couldnt quite get it to work until the above.  We were so close.  Got it working now.  Will send up again for review.

**Review by jacobwgillespie — APPROVED** [review]
LGTM 🚀 :shipit:

**Pull request merged**